internal/rpc/otel_interceptor.go

```diff
@@ -0,0 +1,156 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"connectrpc.com/connect"
+	"github.com/TBD54566975/ftl/internal/log"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	otelFtlRequestKey = "ftl.requestKey"
+	otelFtlVerbRef    = "ftl.verb.ref"
+	otelFtlVerbModule = "ftl.verb.module"
+	otelFtlEventName  = "ftl.message"
+)
```

Comment:
Niiiice! For reference, Jon's [PR from yesterday](https://github.com/TBD54566975/ftl/pull/2140) added a few attributes wrapped in functions. I really like your way of doing const keys here since these attributes all have type string, and this is simpler/shorter than doing functions here, but we should clearly define the pattern at some point so it's clear when to follow/use each. Possibly: for interceptor trace attributes, which are kind of their own thing, we follow this pattern (const keys), but for metrics scattered across many directories, we'll use the functions from the linked PR?

Reply:
Ahh I didn't see Jon's go in. I don't mind using that function pattern at all, happy to switch it, the consistency might be nice

Reply:
We ended up changing his file back to consts so what you have here is perfect :D